components/vc/front/src/torque.c

```diff
@@ -228,75 +399,26 @@ static void torque_init(void)
-    float32_t torque_request_max = torque_data.torque_request_max;
-    float32_t torque = 0.0f;
-
-    const bool gear_change_was_requested = torque_data.gear_change_active;
-    torque_data.gear_change_active = (CANRX_get_signal(VEH, SWS_requestReverse, &gear_change_request) != CANRX_MESSAGE_SNA) &&
-                                     (gear_change_request == CAN_DIGITALSTATUS_ON);
-    const bool gear_change_rising = !gear_change_was_requested && torque_data.gear_change_active;
-
-    const bool race_mode_change_was_requested = torque_data.race_mode_change_active;
-    torque_data.race_mode_change_active = (CANRX_get_signal(VEH, SWS_requestRaceMode, &race_mode_change_requested) != CANRX_MESSAGE_SNA) &&
-                                          (race_mode_change_requested == CAN_DIGITALSTATUS_ON);
-    const bool race_mode_change_rising = !race_mode_change_was_requested && torque_data.race_mode_change_active;
-
-    CAN_digitalStatus_E torque_change_request = CAN_DIGITALSTATUS_SNA;
-    const bool torque_inc_active = (CANRX_get_signal(VEH, SWS_requestTorqueInc, &torque_change_request) != CANRX_MESSAGE_SNA) &&
-                                   (torque_change_request == CAN_DIGITALSTATUS_ON);
-    const bool torque_dec_active = (CANRX_get_signal(VEH, SWS_requestTorqueDec, &torque_change_request) != CANRX_MESSAGE_SNA) &&
-                                   (torque_change_request == CAN_DIGITALSTATUS_ON);
-
-#if FEATURE_IS_ENABLED(FEATURE_REVERSE)
-    if (gear_change_rising)
-    {
-        torque_data.gear = torque_data.gear == GEAR_F ? GEAR_R : GEAR_F;
-    }
-#endif
-
-    if (race_mode_change_rising)
-    {
-        torque_data.race_mode = torque_data.race_mode == RACEMODE_ENABLED ? RACEMODE_PIT : RACEMODE_ENABLED;
-    }
+    const float32_t accelerator_position = apps_getPedalPosition();
+    const float32_t brake_position = bppc_getPedalPosition();
 
     torque_data.state = app_vehicleState_getState() == VEHICLESTATE_TS_RUN ? TORQUE_ACTIVE : TORQUE_INACTIVE;
+    evaluate_gear_change();
+    evaluate_mode_change();
+    evaluate_launch_control(accelerator_position, brake_position);
 
-    if (torque_inc_active ^ torque_dec_active)
-    {
-        const drv_timer_state_E timer_state = drv_timer_getState(&torque_data.torque_change_timer);
-        if (timer_state == DRV_TIMER_STOPPED)
-        {
-            drv_timer_start(&torque_data.torque_change_timer, TORQUE_CHANGE_DELAY);
-            torque_request_max = torque_inc_active ? torque_request_max + 1 : torque_request_max - 1;
-            torque_request_max = SATURATE(MIN_TORQUE_RANGE, torque_request_max, ABSOLUTE_MAX_TORQUE);
-        }
-        else if (timer_state == DRV_TIMER_EXPIRED)
-        {
-            drv_timer_stop(&torque_data.torque_change_timer);
-        }
-
-        torque_data.torque_request_max = torque_request_max;
-    }
-    else
-    {
-        drv_timer_stop(&torque_data.torque_change_timer);
-    }
-
+    float32_t torque_request_max = evaluate_torque_max();
```

Comment:
should this come before evaluating launch control?

Reply:
It doesnt really matter here, there will be logic after max torque setting to take the launch control torque modulation, and then transform that relative to some traction algorithm